util/topsql/reporter/reporter.go

```diff
@@ -464,31 +561,52 @@ type collectedData struct {
 	normalizedPlanMap *sync.Map
 }
 
-// reportData contains data that reporter sends to the agent
-type reportData struct {
-	// collectedData contains the topN collected records and the `others` record which aggregation all records that is out of Top N.
-	collectedData     []*dataPoints
-	normalizedSQLMap  *sync.Map
-	normalizedPlanMap *sync.Map
+// ReportData contains data that reporter sends to the agent
+type ReportData struct {
+	// CPUTimeRecords contains the topN collected records and the `others` record which aggregation all records that is out of Top N.
+	CPUTimeRecords []*tipb.CPUTimeRecord
+	SQLMetas       []*tipb.SQLMeta
+	PlanMetas      []*tipb.PlanMeta
```

Comment:
Why change this? convert from `map` to `slice` will have some memory copy? 
Especially, after https://github.com/pingcap/tidb/pull/27050 , we need report all SQL/Plan meta.

Reply:
1. If there are more than one datasinks, this work can reduce redundant the plan decoding.
2. Here I aim to convert datapoints to tipb shape, don't want to remain this logic over many places, i.e. SingleTargetDataSink.Send, PubSubDataSink.Send ...